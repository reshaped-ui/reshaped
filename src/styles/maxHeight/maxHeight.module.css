@responsive [style*="--rs-max-h:"] {
	@variable --rs-max-h none;
}

@responsive .--type {
	@value literal {
		max-height: var(--rs-max-h) !important;
	}

	@value unit {
		max-height: calc(var(--rs-max-h) * var(--rs-unit-x1)) !important;
	}
}
