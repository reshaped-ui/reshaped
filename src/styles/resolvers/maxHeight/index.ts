import { responsiveClassNames, responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import s from "./maxHeight.module.css";

const getMaxHeightStyles: T.StyleResolver<T.MaxHeight> = (value) => {
	if (!value) return {};
	const variables = responsiveVariables("--rs-max-h", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "--type-unit" : "--type-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames: [s.root, classNames], variables };
};

export default getMaxHeightStyles;
