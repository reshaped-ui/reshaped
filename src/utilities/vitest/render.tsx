import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
