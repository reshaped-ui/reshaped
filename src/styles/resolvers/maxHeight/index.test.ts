import { expect, test, describe } from "vitest";
import maxHeight from "./index";

describe("Styles/MaxHeight", () => {
	test("handles px value", () => {
		expect(maxHeight("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(maxHeight("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(maxHeight(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(maxHeight()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(maxHeight({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
