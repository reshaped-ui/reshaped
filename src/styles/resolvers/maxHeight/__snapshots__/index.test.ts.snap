// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MaxHeight > handles % value 1`] = `
{
  "classNames": [
    "_root_bbb71a",
    [
      "_--type-literal_bbb71a",
    ],
  ],
  "variables": {
    "--rs-max-h-s": "50%",
  },
}
`;

exports[`Styles/MaxHeight > handles px value 1`] = `
{
  "classNames": [
    "_root_bbb71a",
    [
      "_--type-literal_bbb71a",
    ],
  ],
  "variables": {
    "--rs-max-h-s": "50px",
  },
}
`;

exports[`Styles/MaxHeight > handles responsive value 1`] = `
{
  "classNames": [
    "_root_bbb71a",
    [
      "_--type-literal_bbb71a",
      "_--type-literal--l_bbb71a",
    ],
  ],
  "variables": {
    "--rs-max-h-l": "50%",
    "--rs-max-h-s": "50px",
  },
}
`;

exports[`Styles/MaxHeight > handles undefined value 1`] = `null`;

exports[`Styles/MaxHeight > handles unit value 1`] = `
{
  "classNames": [
    "_root_bbb71a",
    [
      "_--type-unit_bbb71a",
    ],
  ],
  "variables": {
    "--rs-max-h-s": 5,
  },
}
`;
